test/e2e_node/hugepages_test.go

```diff
@@ -0,0 +1,248 @@
+	framework.ExpectNoError(err)
+	pageSize, err := strconv.Atoi(strings.TrimSpace(string(outData)))
+	framework.ExpectNoError(err)
+	return pageSize == 2048
+}
+
+// pollResourceAsString polls for a specified resource and capacity from node
+func pollResourceAsString(f *framework.Framework, resourceName string) string {
+	node, err := f.ClientSet.CoreV1().Nodes().Get(framework.TestContext.NodeName, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	amount := amountOfResourceAsString(node, resourceName)
+	framework.Logf("amount of %v: %v", resourceName, amount)
+	return amount
+}
+
+// amountOfResourceAsString returns the amount of resourceName advertised by a node
+func amountOfResourceAsString(node *apiv1.Node, resourceName string) string {
+	val, ok := node.Status.Capacity[apiv1.ResourceName(resourceName)]
+	if !ok {
+		return ""
+	}
+	return val.String()
+}
+
+func runHugePagesTests(f *framework.Framework) {
+	It("should assign hugepages as expected based on the Pod spec", func() {
+		By("by running a G pod that requests hugepages")
+		pod := f.PodClient().Create(&apiv1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pod" + string(uuid.NewUUID()),
+				Namespace: f.Namespace.Name,
+			},
+			Spec: apiv1.PodSpec{
+				Containers: []apiv1.Container{
+					{
+						Image: framework.GetPauseImageName(f.ClientSet),
+						Name:  "container" + string(uuid.NewUUID()),
+						Resources: apiv1.ResourceRequirements{
+							Limits: apiv1.ResourceList{
+								apiv1.ResourceName("cpu"):           resource.MustParse("10m"),
+								apiv1.ResourceName("memory"):        resource.MustParse("100Mi"),
+								apiv1.ResourceName("hugepages-2Mi"): resource.MustParse("50Mi"),
+							},
+						},
+					},
+				},
+			},
+		})
+		podUID := string(pod.UID)
+		By("checking if the expected hugetlb settings were applied")
+		verifyPod := makePodToVerifyHugePages(cm.CgroupName("pod"+podUID), resource.MustParse("50Mi"))
+		f.PodClient().Create(verifyPod)
+		err := framework.WaitForPodSuccessInNamespace(f.ClientSet, verifyPod.Name, f.Namespace.Name)
+		Expect(err).NotTo(HaveOccurred())
+	})
+}
+
+// Serial because the test updates kubelet configuration.
+var _ = SIGDescribe("HugePages [Serial] [Feature:HugePages]", func() {
+	f := framework.NewDefaultFramework("hugepages-test")
+
+	Context("With config updated with hugepages feature enabled", func() {
+		var oldCfg *kubeletconfig.KubeletConfiguration
+
+		BeforeEach(func() {
+			By("verifying hugepages are supported")
+			if !isHugePageSupported() {
+				framework.Skipf("skipping test because hugepages are not supported")
+				return
+			}
+			By("configuring the host to reserve a number of pre-allocated hugepages")
+			Eventually(func() error {
+				err := configureHugePages()
+				if err != nil {
+					return err
+				}
+				return nil
+			}, 30*time.Second, framework.Poll).Should(BeNil())
+			By("enabling hugepages in kubelet")
```

Comment:
@derekwaynecarr would the `tempSetCurrentKubeletConfig` stuff useful here? seems to be used in many places (http://dims.dynu.net/hound/?q=tempSetCurrentKubeletConfig&i=nope&files=&repos=kubernetes)

Reply:
i had tried to look at using that, but ran into ordering problems with multiple BeforeEach clauses.

Reply:
ack thanks!